api/s2n.h

```diff
@@ -871,6 +871,21 @@ extern int s2n_config_add_pem_to_trust_store(struct s2n_config *config, const ch
 S2N_API
 extern int s2n_config_wipe_trust_store(struct s2n_config *config);
 
+/**
+ * Toggle signature verification.
+ *
+ * Although signatures produced by the underlying libcrypto should always be valid,
+ * hardware faults, bugs in the signing implementation, or other uncommon factors
+ * can cause unexpected mistakes in the final signatures. Because these mistakes
+ * can leak information about the private key, applications with low trust in their
+ * hardware or libcrypto may want to verify signatures before sending them.
+ *
+ * However, this feature will significantly impact handshake latency.
+ * Additionally, most libcrypto implementations already check for common errors in signatures.
+ */
+S2N_API
+extern int s2n_config_set_signature_verification(struct s2n_config *config, bool enable);
```

Comment:
I'm not sure the name captures the idea. You could interpret it as enabling/disabling signature verification from the peer, which we definitely don't want 😁.

Maybe something like `verify_after_sign`?